File: mac/Classes/Model/Project/LRProjectFile.m

#import "LRProjectFile.h"
#import "Project.h"

@implementation LRProjectFile

- (id)initWithRelativePath:(NSString *)relativePath project:(Project *)project {
    self = [super init];
    if (self) {
        _relativePath = [relativePath copy];
        _project = project;
    }
    return self;
}

+ (LRProjectFile *)fileWithRelativePath:(NSString *)relativePath project:(Project *)project {
    return [[[self class] alloc] initWithRelativePath:relativePath project:project];
}

- (NSString *)absolutePath {
    return [_project.path stringByAppendingPathComponent:_relativePath];
}

- (NSURL *)absoluteURL {
    return [NSURL fileURLWithPath:self.absolutePath];
}

- (BOOL)exists {
    return [self.absoluteURL checkResourceIsReachableAndReturnError:NULL];
}

@end

File: mac/Classes/LRProjectFile.swift
import Foundation

class LRProjectFile: NSObject {

    let relativePath: String
    let project: Project

    init(relativePath: String, project: Project) {
        self.relativePath = relativePath
        self.project = project
    }

    var absoluteURL: NSURL {
        let rootURL: NSURL = project.rootURL
        return rootURL.URLByAppendingPathComponent(relativePath)
    }

    var absolutePath: String {
        return absoluteURL.path
    }

    var exists: Bool {
        return absoluteURL.checkResourceIsReachableAndReturnError(nil)
    }

    override var description: String {
        return "File<\(relativePath)"
    }

}
